File: COMPILADOR/kokoslan_/src/src/java/ast/KoKoCall.java

package src.java.ast;
import java.util.*;
import java.io.*;
import src.java.eval.KoKoContext;
import src.java.eval.KoKoValue;
import src.java.exception.KoKoEvalException;
public class KoKoCall 
        implements KoKoAst{
    protected KoKoList args;
    protected KoKoAst head;

    public KoKoCall(KoKoAst head,KoKoList args) {
        this.args = args;
        this.head = head;
    }

    public KoKoCall(KoKoAst head) {
        this(head , new KoKoList());
    }

    @Override
    public void genCode(PrintStream out) {
        this.head.genCode(out);
        out.print("(");
        this.args.genCode(out);
        out.print(")");
    }

    @Override
    public KoKoValue eval(KoKoContext ctx) {
        throw new KoKoEvalException("KoKoCall: eval not implemented ");
    }
    
}

File: COMPILADOR/kokoslan/src/KoKoKotlin/ast/KoKoCall.kt
package KoKoKotlin.ast

import KoKoKotlin.eval.KoKoContext
import KoKoKotlin.eval.KoKoValue
import KoKoKotlin.exception.KoKoEvalException
import java.util.*
import java.io.*

import src.java.ast.KoKoAst



class KoKoCall @JvmOverloads constructor(protected var head: KoKoAst, protected var args: KoKoList = KoKoList()) : KoKoAst {

    override fun genCode(out: PrintStream) {
        this.head.genCode(out)
        out.print("(")
        this.args.genCode(out)
        out.print(")")
    }

    override fun eval(ctx: KoKoContext): KoKoValue {
        throw KoKoEvalException("KoKoCall: eval not implemented ")
    }

}
